test/testdrive/protobuf.td

```diff
@@ -96,8 +96,12 @@ $ kafka-ingest format=protobuf topic=messages-partitioned message=struct timesta
   WITH (start_offset=[1,0])
   FORMAT PROTOBUF MESSAGE '.Struct' USING SCHEMA FILE '${testdrive.protobuf-descriptors-file}'
 
-! CREATE MATERIALIZED SOURCE protomessages_partitioned FROM
-  KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-messages-partitioned-${testdrive.seed}'
-  WITH (start_offset=[1,0])
+$ kafka-create-topic topic=tester
+
+$ kafka-ingest topic=tester format=protobuf message=simpleid schema='syntax = "proto3"; message SimpleId { string id = 1; }' publish=true
```

Comment:
So, stuffing the schema into the argument value is going to get pretty hairy pretty fast. Here's my proposal: teach `$ set` to accept multiline input. I think it shakes out something like this:

```diff
diff --git a/doc/developer/testdrive.md b/doc/developer/testdrive.md
index a623ccf38..f7b9ca69c 100644
--- a/doc/developer/testdrive.md
+++ b/doc/developer/testdrive.md
@@ -369,6 +369,19 @@ $ set schema={
 
+#### `$ set name`
+
+Sets the variable to the possibly multi-line value that follows.
+
+```
+$ set schema
+syntax = "proto3";
+
+message SimpleId {
+  string id = 1;
+}
+```
+
 ## Referencing variables
 
 #### `${variable_name}`
diff --git a/src/testdrive/src/action.rs b/src/testdrive/src/action.rs
index 524aa12d9..1925f6b83 100644
--- a/src/testdrive/src/action.rs
+++ b/src/testdrive/src/action.rs
@@ -488,7 +488,13 @@ pub fn build(cmds: Vec<PosCommand>, state: &State) -> Result<Vec<PosAction>, Err
                         Box::new(sleep::build_sleep(builtin).map_err(wrap_err)?)
                     }
                     "set" => {
-                        vars.extend(builtin.args);
+                        for (key, val) in builtin.args {
+                            if val.is_empty() {
+                                vars.insert(key, builtin.input.join("\n"));
+                            } else {
+                                vars.insert(key, val);
+                            }
+                        }
                         continue;
                     }
                     _ => {
diff --git a/src/testdrive/src/parser.rs b/src/testdrive/src/parser.rs
index 1b613f20e..16ef486ca 100644
--- a/src/testdrive/src/parser.rs
+++ b/src/testdrive/src/parser.rs
@@ -103,9 +103,11 @@ fn parse_builtin(line_reader: &mut LineReader) -> Result<BuiltinCommand, InputEr
     let mut args = HashMap::new();
     for el in builtin_reader {
         let (pos, token) = el?;
-        let pieces = match token.split_once('=') {
-            Some((key, value)) => vec![key, value],
-            None => {
+        let pieces: Vec<_> = token.splitn(2, '=').collect();
+        let pieces = match pieces.as_slice() {
+            [key, value] => vec![*key, *value],
+            [key] => vec![*key, ""],
+            _ => {
                 return Err(InputError {
                     msg: "command argument is not in required key=value format".into(),
                     pos,
diff --git a/test/testdrive/protobuf.td b/test/testdrive/protobuf.td
index 4ad227304..5481624bc 100644
--- a/test/testdrive/protobuf.td
+++ b/test/testdrive/protobuf.td
@@ -98,7 +98,14 @@ $ kafka-ingest format=protobuf topic=messages-partitioned message=struct timesta
 
 $ kafka-create-topic topic=tester
 
-$ kafka-ingest topic=tester format=protobuf message=simpleid schema='syntax = "proto3"; message SimpleId { string id = 1; }' publish=true
+$ set schema
+syntax = "proto3";
+
+message SimpleId {
+  string id = 1;
+}
+
+$ kafka-ingest topic=tester format=protobuf message=simpleid schema=${schema} publish=true
 {"id": "a"}
 
 ! CREATE MATERIALIZED SOURCE tester

```

Reply:
We probably also want to find a way to avoid duplicating the schemas in test/testdrive/*.td and the .proto files in src/testdrive/format/protobuf, but not seeing any immediately obvious ways to make that happen.